### PR TITLE
[STORM-1594] org.apache.storm.tuple.Fields can throw NPE if given invalid field in selector

### PR DESCRIPTION
**Pull request by lujinhong**
fix STORM-1594.

### CODE REVIEW

File: storm-core/src/jvm/org/apache/storm/tuple/Fields.java [hunk 1]
```diff
@@ -50,8 +50,12 @@ public Fields(List<String> fields) {
     
     public List<Object> select(Fields selector, List<Object> tuple) {
         List<Object> ret = new ArrayList<>(selector.size());
-        for(String s: selector) {
-            ret.add(tuple.get(_index.get(s)));
```
Comment (abhishekagarwal87) [inline storm-core/src/jvm/org/apache/storm/tuple/Fields.java]: You can replace _index.get(s) with fieldIndex(s) which throws IllegalArgumentException.

Comment (lujinhong) [inline storm-core/src/jvm/org/apache/storm/tuple/Fields.java]: yes, it is a better idea.

**Comment by abhishekagarwal87** [comment]
Thank you @lujinhong. I am +1 pending travis. Please squash the commits once travis is passed.

**Comment by HeartSaVioR** [comment]
+1 Travis error is unrelated.
@lujinhong Could you squash the commits, or if you mind I can do that for merging step. Let me know which way do you prefer.

**Comment by abellina** [comment]
Looks good @lujinhong. Thanks!

**Comment by revans2** [comment]
@HeartSaVioR looks like one of the test failures was related.

https://issues.apache.org/jira/browse/STORM-2022

**Comment by HeartSaVioR** [comment]
@revans2 Yes I missed that. My bad. Thanks for finding.

**Pull request closed without merging**